03week/towersOfHanoi.js

```diff
@@ -13,37 +54,70 @@ let stacks = {
   c: []
 };
 
-function printStacks() {
+const availableChoices = ['a', 'b', 'c'];
+
+
+const printStacks = () => {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+const movePiece = (startStack, endStack) => {
+  let blockToMove = stacks[startStack].pop();
+  stacks[endStack].push(blockToMove);
 }
 
-function isLegal() {
-  // Your code here
+const isLegal = (endStack, topBlock, bottomBlock) => {
+  if (stacks[endStack].length === 0 || topBlock < bottomBlock) {
+    return true;
+  } else {
+    return false;
+  }
+}
 
+const checkForWin = () => {
+  if (stacks.c.length === 4) {
+    console.log('You won!');
+    return true;
+  } else {
+    return false;
+  }
 }
 
-function checkForWin() {
-  // Your code here
+const towersOfHanoi = (startStack, endStack) => {
+  startStack = startStack.trim().toLowerCase();
+  endStack = endStack.trim().toLowerCase();
 
+  if (validInput(startStack, endStack)) {
+    let topBlock = stacks[startStack][stacks[startStack].length - 1];
+    let bottomBlock = stacks[endStack][stacks[endStack].length - 1];
+    if(isLegal(endStack, topBlock, bottomBlock)) {
+      movePiece(startStack, endStack);
+    } else {
+      console.log('Illegal move!');
+    }
+  } else {
+    console.log('Invalid input!');
+  }
 }
 
-function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+const validInput = (startStack, endStack) => {
+  if(availableChoices.indexOf(startStack) !== -1 && availableChoices.indexOf(endStack) !== -1) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
-function getPrompt() {
+const getPrompt = () => {
   printStacks();
   rl.question('start stack: ', (startStack) => {
     rl.question('end stack: ', (endStack) => {
       towersOfHanoi(startStack, endStack);
+      if (checkForWin()) {
```

Comment:
you shouldn't touch this, you only want to check for win if there's a valid move. This should be handled in the towersOfHanoi() as instructed